File: algorithms/0590.n-ary-tree-postorder-traversal/solution.rb
# Definition for a Node.
# class Node
#     attr_accessor :val, :children
#     def initialize(val)
#         @val = val
#         @children = []
#     end
# end

# @param {Node} root
# @return {List[int]}
def postorder(root)
  res = []

  return res if root.nil?

  root.children.each do |child|
    res.concat postorder(child)
  end

  res << root.val
end

File: algorithms/0590.n-ary-tree-postorder-traversal/solution.go
/**
 * Definition for a Node.
 * type Node struct {
 *     Val int
 *     Children []*Node
 * }
 */
func postorder(root *Node) []int {
	var res []int

	if root == nil {
		return res
	}

	for _, child := range root.Children {
		res = append(res, postorder(child)...)
	}

	res = append(res, root.Val)
	return res
}
